rotkehlchen/chain/ethereum/modules/lido/decoder.py

```diff
@@ -0,0 +1,156 @@
+        collateral_amount = token_normalized_value_decimals(
+            token_amount=amount_raw,
+            token_decimals=18,
+        )
+
+        # Searching for the exact paired stETH token reception for validating the decoding
+        steth_minted_tokens = None
+        # Here searching for the paired stETH mint operation, with a hacky way for the issue:
+        # https://github.com/lidofinance/lido-dao/issues/442
+        for tx_log in context.all_logs:
+            if (
+                tx_log.address == self.steth_evm_address and
+                tx_log.topics[0] == ERC20_OR_ERC721_TRANSFER and
+                hex_or_bytes_to_address(tx_log.topics[1]) == ZERO_ADDRESS and  # from
+                hex_or_bytes_to_address(tx_log.topics[2]) == sender_address and  # to
+                abs(amount_raw - hex_or_bytes_to_int(tx_log.data[:32])) < STETH_MAX_ROUND_ERROR_WEI
+            ):
+                steth_minted_tokens = token_normalized_value_decimals(
+                    token_amount=hex_or_bytes_to_int(tx_log.data[:32]),
+                    token_decimals=18,
+                )
+                break
+        else:  # did not break/find anything
+            log.error(
+                f'At lido steth submit decoding of tx {context.transaction.tx_hash.hex()}'
+                f'did not find the related stETH token generation',
+            )
+            return DEFAULT_DECODING_OUTPUT
+
+        # Searching for the already decoded event,
+        # containing the ETH transfer of the submit transaction
+        paired_event = None
+        for event in context.decoded_events:
+            if (
+                event.address == self.steth_evm_address and
+                event.asset == A_ETH and
+                event.balance.amount == collateral_amount and
+                event.event_type == HistoryEventType.SPEND and
+                event.location_label == sender_address
+            ):
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.notes = f'Submit {collateral_amount} ETH to Lido for receiving stETH in exchange'  # noqa: E501
+                event.counterparty = CPT_LIDO_ETH
+                #  preparing next action to be processed when erc20 transfer will be decoded by rotki  # noqa: E501
+                #  needed because submit levent is emitted prior of erc20 transfer, so it is not decoded yet  # noqa: E501
+                # TODO: to be confirmed with ROTKI team if it is not possible to have the erc20 event available before this decoder is called  # noqa: E501
+                paired_event = event
+                action_from_event_type = HistoryEventType.RECEIVE
+                action_to_event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                action_to_notes = f'Receive {steth_minted_tokens} stETH in exchange of the deposited ETH'  # noqa: E501
```

Comment:
So you will probably need an extra action in the `ActionItem` processing in evm/decoding/decoder.py that if there is an amount tolerance check to change the amount in the notes when matching happens.

So here: https://github.com/rotki/rotki/blob/664484982a3b2eb23f9afbe2e12688d67e153840/rotkehlchen/chain/evm/decoding/decoder.py#L923

(pseudocode)

```python
                if action_item.to_notes is not None:
                    transfer.notes = action_item.to_notes if action_item.amount_error_tolerance is None else action_item.to_notes.format(amount=transfer.balance.amount)
```

effectively making it take the correct amount, which is the amount of the matched ERC20 token transfer.